python/cudf/cudf/tests/test_avro.py

```diff
@@ -49,16 +50,29 @@ def _make_path_or_buf(src):
 @pytest.mark.parametrize("inputfile, columns", [("example.avro", None)])
 def test_avro_reader_basic(datadir, inputfile, columns, engine):
     path = datadir / inputfile
+
     try:
-        reader = fa.reader(open(path, "rb"))
+        avro_file = open(path, "rb")
+        reader = fa.reader(avro_file)
+
     except Exception as excpr:
         if type(excpr).__name__ == "FileNotFoundError":
             pytest.skip(".avro file is not found")
         else:
             print(type(excpr).__name__)
 
     expect = pd.DataFrame.from_records(reader)
-    got = cudf.read_avro(path, engine=engine, columns=columns)
+
+    if not avro_file.closed:
+        avro_file.close()
+
+    # `read_avro()` is deliberately called twice. I have found inconsistent
+    # file access behavior dropping the first or second item in the first
+    # row on the first column read. Consistently producing. Accessint twice
+    # at runtime ensures data is loaded correctly.
+    got = reorder_dataframe_columns_to_match_pandas(
+        gdf=cudf.read_avro(path, engine=engine, columns=columns), pdf=expect
+    )
```

Comment:
I don't see `read_avro()` being called twice here. Stale comment? Does an issue need to be filed for this dropped first or second item issue?

Reply:
Oh, I think I actually dropped the call, and I think that might be why that test fails some of the times as below.

Reply:
PR #3588 may have fixed the Avro issues. Testing that now.